sasmodels/kernelpy.py

```diff
@@ -309,6 +324,22 @@ def vector_Iq(q, *args):
         vector_Iq.vectorized = True
         model_info.Iq = vector_Iq
 
+def _create_vector_Fq(model_info):
+    """
+    Define Fq as a vector function if it exists.
+    """
+    # Note that this is doing slightly too much work since we are composing
+    # two vector results which are then interlaced via _pack_fq above.
+    Fq = model_info.Fq
+    if callable(Fq) and not getattr(Fq, 'vectorized', False):
```

Comment:
I take it you're using `callable` here because `model_info.compiled` isn't exactly what you want. If it is, consider being consistent with the other changes.

Reply:
`info.compiled` is for `info.Fq` and `info.Iq` jointly.  Only one of them will be defined, and it may be defined using C source or using a callable python function.

It occurs to me that we could use some numba transforms here so that the code runs as fast as a C model on the CPU (or faster since loop parallelization can run over multiple cores). Python models get much easier to write when we don't have to vectorize across if statements.